### PR TITLE
Ddcnl 3134

### PR DESCRIPTION
**Pull request by Muhammad-Ali-Ahmad**

### CODE REVIEW

File: app/views/CapitalGainsView.scala.html [hunk 1]
```diff
@@ -18,139 +18,164 @@
...
+        <h1 class="heading-xlarge">
+            <span class="gray heading-medium block-heading">
+              @Html(Messages("generic.tax_year_heading", viewModel.taxYearFrom, viewModel.taxYearTo))
+            </span>
+          @Html(Messages("ats.capital_gains_tax.subtitle"))
+        </h1>
+        <p class="lede">@Html(Messages("ats.capital_gains_tax.lede"))</p>
+        @if(!viewModel.totalCapitalGainsTaxAmount.isZero) {
+          <p id="total-cg-description">@Html(Messages("ats.capital_gains_tax.description",
+            """<span id="total-cg-tax-rate">"""+viewModel.totalCgTaxRate.percent+"""</span>"""))</p>
```
Comment (thirlz95) [inline app/views/CapitalGainsView.scala.html:74]: can you use plays StringInterpolation -> https://www.playframework.com/documentation/2.6.x/ScalaTemplates

File: app/views/CapitalGainsView.scala.html [hunk 2]
```diff
@@ -18,139 +18,164 @@
...
+                    <td class="width80">@Html(Messages("ats.capital_gains_tax.table.rcpi", toCurrency(viewModel.rpciHigherTotalAmount), viewModel.rpciHigherRate.percent))</td>
+                    <td id="rpci-higher-rate-amount" class="width20">@currencyAmountView(viewModel.rpciHigherTax)</td>
+                </tr>
+              }
+
+
+              @if(!viewModel.entrepreneursReliefRateBefore.isZero) {
+              <tr id="entrepreneurs-relief-rate-section">
+                <td>@Html(Messages("ats.capital_gains_tax.table.line4",
+                    """<span id="entrepreneurs-relief-rate-before">"""+currencyAmountView(viewModel.entrepreneursReliefRateBefore)+"""</span>""",
```
Comment (thirlz95) [inline app/views/CapitalGainsView.scala.html:133]: as above check for using StringInterpolation from twirl

File: app/views/CapitalGainsView.scala.html [hunk 3]
```diff
@@ -18,139 +18,164 @@
...
+                    """<span id="entrepreneurs-relief-rate">""" + viewModel.entrepreneursReliefRate + """</span>"""))</td>
+                <td id="entrepreneurs-relief-rate-amount">@currencyAmountView(viewModel.entrepreneursReliefRateAmount)</td>
+              </tr>
+              }
+
+              @if(!viewModel.ordinaryRateBefore.isZero) {
+              <tr id="ordinary-rate-section">
+                <td>@Html(Messages("ats.capital_gains_tax.table.line5",
+                    """<span id="ordinary-rate-before">"""+currencyAmountView(viewModel.ordinaryRateBefore)+"""</span>""",
```
Comment (thirlz95) [inline app/views/CapitalGainsView.scala.html:142]: as above check for using StringInterpolation from twirl

File: app/views/CapitalGainsView.scala.html [hunk 4]
```diff
@@ -18,139 +18,164 @@
...
+                    """<span id="ordinary-rate">""" + viewModel.ordinaryRate + """</span>"""))</td>
+                <td id="ordinary-rate-amount">@currencyAmountView(viewModel.ordinaryRateAmount)</td>
+              </tr>
+              }
+
+              @if(!viewModel.upperRateBefore.isZero) {
+              <tr id="upper-rate-section">
+                <td>@Html(Messages("ats.capital_gains_tax.table.line6",
+                    """<span id="upper-rate-before">"""+currencyAmountView(viewModel.upperRateBefore)+"""</span>""",
```
Comment (thirlz95) [inline app/views/CapitalGainsView.scala.html:151]: as above check for using StringInterpolation from twirl

File: app/views/IncomeBeforeTaxView.scala.html [hunk 5]
```diff
@@ -17,106 +17,123 @@
...
+
+        <table>
+          <thead>
+            <tr>
+              <th>@Html(Messages("ats.income_before_tax.table_header"))</th>
+              <th>@Html(Messages("generic.amount_pounds"))</th>
+            </tr>
+          </thead>
+
+          <tbody id="income-before-tax-table">
```
Comment (thirlz95) [inline app/views/IncomeBeforeTaxView.scala.html:78]: more of a question than something to action but couldn't this be rewritten almost using a map/case statments to pass the table rows statement and just pass variables to the first td message, id and which view model to save from having to rewrite each table row

File: app/views/NicsView.scala.html [hunk 6]
```diff
@@ -19,67 +19,85 @@
...
-    </table>
+        <h1 class="heading-xlarge">
+            <span class="gray heading-medium block-heading">
+              @Html(Messages("generic.tax_year_heading", viewModel.taxYearFrom, viewModel.taxYearTo))
+            </span>
+            @Html(Messages("ats.nics.tax_and_nics.title"))
+        </h1>
+      <p class="lede">@Html(Messages("ats.nics.tax_and_nics.lede"))</p>
+        <p>@Html(Messages("ats.nics.tax_and_nics.description",
+          "<span id=\"total-cg-tax-rate\">"+positiveOrZero(viewModel.nicsAndTaxRate).percent+"</span>"))</p>
```
Comment (thirlz95) [inline app/views/NicsView.scala.html:71]: interpolation possibly here?

**Review by thirlz95 — CHANGES_REQUESTED** [review]

File: app/views/CapitalGainsView.scala.html (hunk 2 above)
Comment (thirlz95) [reply app/views/CapitalGainsView.scala.html:133]: Resolving as ticket is just covering template upgrade

File: app/views/CapitalGainsView.scala.html (hunk 3 above)
Comment (thirlz95) [reply app/views/CapitalGainsView.scala.html:142]: Resolving as ticket is just covering template upgrade

File: app/views/IncomeBeforeTaxView.scala.html (hunk 5 above)
Comment (thirlz95) [reply app/views/IncomeBeforeTaxView.scala.html:78]: Resolving as ticket is just covering template upgrade

File: app/views/CapitalGainsView.scala.html (hunk 4 above)
Comment (thirlz95) [reply app/views/CapitalGainsView.scala.html:151]: Resolving as ticket is just covering template upgrade

File: app/views/NicsView.scala.html (hunk 6 above)
Comment (thirlz95) [reply app/views/NicsView.scala.html:71]: Resolving as ticket is just covering template upgrade

**Review by thirlz95 — APPROVED** [review]
From a template switch over i've not been able to spot any issues, earlier comments where mention upgrades in the templating but that isn't covered in this ticket.

**Review by shikkhasingh — APPROVED** [review]

**Review by shikkhasingh — APPROVED** [review]

**Pull request merged**